### PR TITLE
Add support for WorldProvider-specific ingame music

### PR DESCRIPTION
**Pull request by williewillus**
Pretty straightforward

### CODE REVIEW

File: patches/minecraft/net/minecraft/world/WorldProvider.java.patch [hunk 1]
```diff
@@ -222,6 +222,19 @@
...
 +    }
++
++    /**
++     * Called on the client to get the music type to play when in this world type.
++     * At the time of calling, the client player is guaranteed to be non-null
++     * @return null to use vanilla logic, otherwise a MusicType to play in this world
++     */
++    @Nullable
++    @SideOnly(Side.CLIENT)
++    public net.minecraft.client.audio.MusicTicker.MusicType getMusic()
```
Comment (bs2609) [inline patches/minecraft/net/minecraft/world/WorldProvider.java.patch]: How about `getMusicType` for the name here?

**Comment by williewillus** [comment]
renamed

**Comment by bs2609** [comment]
Needs an update for merge conflicts.

**Comment by williewillus** [comment]
conflicts resolved

**Comment by Waterpicker** [comment]
I'm one of the developers working on the modern rewrite of the mod Dimensional Doors. The custom dimension limbo from it has its own music. I'm having issues figuring out how to make it only do that. This PR adds exactly what I need. I hope it gets added soon.

**Comment by LexManos** [comment]
Oh lovely more enums that need to be de-enumed.

**Comment by Waterpicker** [comment]
I unfortunately have to comment about something missing from the PR. The entry in clientTypes variable is missing an entry for MusicType.

**Comment by williewillus** [comment]
https://github.com/MinecraftForge/MinecraftForge/pull/4682

**Pull request merged**